ansible_base/lib/testing/fixtures.py

```diff
@@ -130,16 +130,19 @@ def expected_log(no_log_messages):
 
     @contextmanager
     def f(patch, severity, substr, assert_not_called=False):
-        with mock.patch(patch) as logger:
+        with mock.patch(f'{patch}.{severity}') as logger:
             with no_log_messages():
                 yield
-            sev_logger = getattr(logger, severity)
+
+            call_count = 0
+            for call in logger.call_args_list:
+                if substr in call.args[0]:
+                    call_count = call_count + 1
```

Comment:
```suggestion
                    call_count += 1
```

or just

call_count = len([call for call in logger.call_args_list if substr in call.args[0]])

Reply:
No need to construct a temporary list that is immediately discarded.
```
call_count = sum(1 for call in logger.call_args_list if substr in call.args[0])
```

Reply:
Ah, `sum()` with 1 works... I originally left out the list, but count() won't work on a generator so I changed it.